ttnn/cpp/ttnn/operations/ccl/barrier/device/kernels/barrier_sender_HPerf_LPrec.cpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+#include "tt_metal/hw/inc/dataflow_api.h"
+#include "debug/assert.h"
+#include "debug/dprint.h"
+#include <array>
+#include "tt_metal/hw/inc/ethernet/dataflow_api.h"
+
+
+struct addr_sem_pair {
+    uint32_t addr;
+    uint32_t sem_addr;
+};
+
+
+static constexpr bool DISABLE_CONTEXT_SWITCHING = true;
+static constexpr uint8_t NUM_CHANNELS = 8;
+
+FORCE_INLINE void eth_setup_handshake(uint32_t handshake_register_address) {
```

Comment:
Same handshake comment as from receiver (but you can use the sender side functions instead)